Assets/Scripts/Models/TileType.cs

```diff
@@ -6,87 +6,78 @@
 
     private static TileType empty;
     private static TileType floor;
 
-    // Base cost of pathfinding over this tile, movement cost and any furniture will modify the effective value
-    private float pathfindingWeight = 1f;
-
-    // Additional cost of pathfinding over this tile, will be added to pathfindingWeight * MovementCost
-    private float pathfindingModifier = 0f;
+    private Job buildingJob;
 
-    // Will this even be needed?
-    private TileType(string name, string description, float baseMovementCost)
+    private TileType()
     {
-        this.Name = name;
-        this.Description = description;
-        this.BaseMovementCost = baseMovementCost;
-
-        // Add this to the dictionary of all tileTypes
-        tileTypeDictionary[name] = this;
+        PathfindingModifier = 0.0f;
+        PathfindingWeight = 1.0f;
     }
 
-    private TileType()
+    public static TileType Empty
     {
+        get { return empty ?? (empty = TileTypes["Empty"]); }
     }
 
-    // Just two util functions to not break every link to TileType.(Empty | Floor)
-    public static TileType Empty 
+    public static TileType Floor
     {
-        get { return empty ?? (empty = tileTypeDictionary["Empty"]); } 
+        get { return floor ?? (floor = TileTypes["Floor"]); } 
     }
 
-    public static TileType Floor 
+    /// <summary>
+    /// All currently registerd TileTypes.
+    /// </summary>
+    public static TileType[] LoadedTileTypes
     {
-        get { return floor ?? (floor = tileTypeDictionary["Floor"]); } 
+        get
+        {
+            return TileTypes.Values.ToArray();
+        }
     }
 
-    public string Type { get; protected set; }
+    /// <summary>
+    /// Unique TileType identifier.
+    /// </summary>
+    public string Type { get; private set; }
 
-    public string Name { get; protected set; }
+    public string Name { get; private set; }
 
-    public string Description { get; protected set; }
+    public string Description { get; private set; }
 
-    public float BaseMovementCost { get; protected set; }
+    public float BaseMovementCost { get; private set; }
 
     /// <summary>
     /// Gets or sets the TileType's pathfinding weight which is multiplied into the Tile's final PathfindingCost.
     /// </summary>
-    public float PathfindingWeight
-    {
-        get { return pathfindingWeight; }
-        set { pathfindingWeight = value; }
-    }
+    public float PathfindingWeight { get; set; }
 
     /// <summary>
     /// Gets or sets the TileType's pathfinding modifier which is added into the Tile's final PathfindingCost.
     /// </summary>
-    public float PathfindingModifier
-    {
-        get { return pathfindingModifier; }
-        set { pathfindingModifier = value; }
-    }
+    public float PathfindingModifier { get; set; }
 
-    // TODO!
     public bool LinksToNeighbours { get; protected set; }
```

Comment:
Should you use private here, and in the next few properties? Is changed above from protected.